zdm/grid.py

```diff
@@ -69,15 +68,16 @@ def __init__(self, survey, state,
 
     def init_luminosity_functions(self):
         if self.luminosity_function==0:  # Power-law
-            self.array_cum_lf=zdm.array_cum_power_law
-            self.vector_cum_lf=zdm.vector_cum_power_law
-            self.array_diff_lf=zdm.array_diff_power_law
-            self.vector_diff_lf=zdm.vector_diff_power_law
+            self.array_cum_lf=energetics.array_cum_power_law
+            self.vector_cum_lf=energetics.vector_cum_power_law
+            self.array_diff_lf=energetics.array_diff_power_law
+            self.vector_diff_lf=energetics.vector_diff_power_law
         elif self.luminosity_function==1:  # Gamma function
-            self.array_cum_lf=zdm.array_cum_gamma
-            self.vector_cum_lf=zdm.vector_cum_gamma
-            self.array_diff_lf=zdm.array_diff_gamma
-            self.vector_diff_lf=zdm.vector_diff_gamma
+            self.array_cum_lf=energetics.array_cum_gamma_spline
+            self.vector_cum_lf=energetics.vector_cum_gamma
```

Comment:
should there not be an equivalent vector_cum_gamma_spline function here? After all, the 2D version is mostly a wrapper around the 1D version